### PR TITLE
Update Conversion of consumer in doc

### PR DESCRIPTION
**Pull request by Zlopez**
Describe the process of conversion to Fedora Messaging for the-new-hotness.

Signed-off-by: Michal Konečný <mkonecny@redhat.com>

### CODE REVIEW

File: docs/tutorial/conversion.rst [hunk 1]
```diff
@@ -296,4 +296,207 @@ Create the database and start the application::
...
+            self.handle_buildsys_scratch(msg)
+        else:
+            _log.debug("Dropping %r %r" % (topic, body))
+            pass
+
+In this case we are working with the message using the standard
+``fedora_messaging.message.Message`` methods. It is always better to use schema specific
+methods for any topic you are receiving.
+
+Testing
```
Comment (jeremycline) [inline docs/tutorial/conversion.rst]: I think it would be better if most of this was replaced with a link to the hotness docs and a sentence like "In a [configured development environment](link to docs), we can easily test our conversion by..."

File: docs/tutorial/conversion.rst [hunk 2]
```diff
@@ -296,4 +296,207 @@ Create the database and start the application::
...
+            consumer=self, config=self.config["bugzilla"]
+        )
+        self.buildsys = hotness.buildsys.Koji(consumer=self, config=self.config["koji"])
+
+        default = "https://pagure.io/releng/fedora-scm-requests"
+        self.repo_url = self.config.get("repo_url", default)
+        default = "https://pdc.fedoraproject.org"
+        self.pdc_url = self.config.get("pdc_url", default)
+        default = "https://src.fedoraproject.org"
+        self.dist_git_url = self.config.get("dist_git_url", default)
```
Comment (jeremycline) [inline docs/tutorial/conversion.rst]: A lot of this stuff is specific to hotness and not related to messaging at all. It might be easier to follow if you removed anything not related to the messaging conversion (with a note that unrelated code is removed) so that readers who aren't familiar with the-new-hotness can quickly see what changes and what's relevant.

**Review by jeremycline — DISMISSED** [review]
I think some of it can be simplified a bit and there's a couple small grammar issues, but overall this is off to a good start.

File: docs/tutorial/conversion.rst [hunk 3]
```diff
@@ -296,4 +296,207 @@ Create the database and start the application::
 Converting consumers
 ====================
 
-TODO the-new-hotness
+.. the-new-hotness
+
+Let's use `the-new-hotness`_ app as an example. Clone the code using the
+following command::
+
+    git clone https://github.com/fedora-infra/the-new-hotness.git
```
[the remark this answers is not in the document]
Comment (Zlopez) [reply docs/tutorial/conversion.rst]: I copied this from publisher examples.

**Comment by Zlopez** [comment]
Is there something else I need to do to get this PR merged?

I think I already addressed all the comments in latest commit.

File: docs/tutorial/installation.rst [hunk 4]
```diff
@@ -32,6 +32,9 @@ Install RabbitMQ and start it::
     dnf install rabbitmq-server
     systemctl start rabbitmq-server
 
+Enable RabbitMQ web admin interface::
```
Comment (abompard) [inline docs/tutorial/installation.rst:37]: Please add a blank line below this, it's required by RST

File: docs/tutorial/conversion.rst [hunk 5]
```diff
@@ -296,4 +296,150 @@ Create the database and start the application::
...
+In ``the-new-hotness`` app, all calls to consume messages on fedmsg are going
+through the ``hotness.consumers.BugzillaTicketFiler.consume`` method. We can
+thus modify this function to make it use Fedora Messaging instead of fedmsg.
+
+Configuration
+-------------
+First we need to convert configuration file from fedmsg format to Fedora
+Messaging. Unlike fedmsg, fedora-messaging does not allow for arbitrary
+configuration keys. You will need to place any application-specific configuration
+in the :ref: dictionary.
```
Comment (abompard) [inline docs/tutorial/conversion.rst]: The backticks are missing here, no link is generated.

File: docs/tutorial/conversion.rst [hunk 6]
```diff
@@ -296,4 +296,150 @@ Create the database and start the application::
...
+    arguments = {}
+
+Any application specific configuration should go to ``consumer_config`` section
+:ref:`config`.
+
+Init method
+-----------
+The ``BugzillaTicketFiler`` class in ``consumers.py`` is doing all the consuming work.
+First we need to change the inheritance of this class. Fedora Messaging consumer class
+needs to inherit only from ``object``.
```
Comment (abompard) [inline docs/tutorial/conversion.rst]: The next version is going to be python3-only, so the class does not need to inherit from anything.

File: docs/tutorial/conversion.rst [hunk 7]
```diff
@@ -296,4 +296,150 @@ Create the database and start the application::
...
+
+Then we need to modify the ``__init__`` method and use the ``fedora_messaging.config.conf``
+dictionary instead of the fedmsg configuration. The ``__init__`` method could look something
+like this after the change::
+
+    from fedora_messaging.config import conf
+
+    class BugzillaTicketFiler(object):
```
Comment (abompard) [inline docs/tutorial/conversion.rst]: Please remove the `object` inheritance.

**Review by abompard — CHANGES_REQUESTED** [review]

File: docs/tutorial/conversion.rst (hunk 5 above)
Comment (Zlopez) [reply docs/tutorial/conversion.rst]: I'm not sure what this sentence should point to, but the dictionary label no longer exist, so I remover the whole sentence from the docs.

File: docs/tutorial/conversion.rst (hunk 6 above)
Comment (Zlopez) [reply docs/tutorial/conversion.rst]: This sentence was removed

**Comment by Zlopez** [comment]
I rebased to latest develop branch and addressed the remaining issues. It should be fine now.

**Review by abompard — APPROVED** [review]

**Comment by abompard** [comment]
Could you please rebase on the `develop` branch? After that it should be all good. Thanks!

**Comment by abompard** [comment]
Nevermind, it's all good, thanks.
